storage/store.go

```diff
@@ -1510,16 +1518,21 @@ func (s *Store) ProposeRaftCommand(idKey cmdIDKey, cmd proto.RaftCommand) <-chan
 	if err != nil {
 		log.Fatal(err)
 	}
-	etr, ok := value.(*proto.EndTransactionRequest)
-	if ok && etr.InternalCommitTrigger != nil &&
-		etr.InternalCommitTrigger.ChangeReplicasTrigger != nil {
-		// EndTransactionRequest with a ChangeReplicasTrigger is special because raft
-		// needs to understand it; it cannot simply be an opaque command.
-		crt := etr.InternalCommitTrigger.ChangeReplicasTrigger
-		return s.multiraft.ChangeGroupMembership(cmd.RangeID, string(idKey),
-			changeTypeInternalToRaft[crt.ChangeType],
-			proto.MakeRaftNodeID(crt.NodeID, crt.StoreID),
-			data)
+	for _, union := range cmd.Cmd.Requests {
+		args := union.GetValue().(proto.Request)
+		etr, ok := args.(*proto.EndTransactionRequest)
+		if ok && etr.InternalCommitTrigger != nil && etr.InternalCommitTrigger.ChangeReplicasTrigger != nil {
+			if len(cmd.Cmd.Requests) != 1 {
+				panic("EndTransaction should only ever occur by itself in a batch")
```

Comment:
yes, this is too much. Not sure why @spencerkimball put this in here - maybe there's a caveat in the work so far we're not seeing? Added a comment, this'll be massaged away with #2141 soon anyway.